### PR TITLE
Add read/write support for fields relevant to hexagonal tile layers.

### PR DESCRIPTION
**Pull request by BorisTheBrave**
Also fix some serialization bugs caused by unexpected XML.

### CODE REVIEW

File: TiledLib/TmxMisc.cs [hunk 1]
```diff
@@ -60,6 +60,7 @@ public static void ReadTileset(this XmlReader reader, Tileset ts)
                         reader.ReadTile(ts.TileProperties, ts.TileAnimations);
                         break;
                     default:
+                        reader.Skip();
```
Comment (BorisTheBrave) [inline TiledLib/TmxMisc.cs:69]: This was causing loads to hang when encountering unexpected elements, such as the terrain element.

File: TiledLib/TmxMisc.cs [hunk 2]
```diff
@@ -79,25 +80,32 @@ static void ReadTile(this XmlReader reader, Dictionary<int, Dictionary<string, s
...
-                        tileAnimations[id] = reader.ReadAnimation();
-                        break;
-                    default:
-                        reader.Skip(); //TODO: Add logging.
-                        break;
-                }
-
-            if (reader.Name == "tile")
-                reader.ReadEndElement();
+            if (reader.IsEmptyElement)
```
Comment (BorisTheBrave) [inline TiledLib/TmxMisc.cs:89]: Some of the sample files I had have xml like:

    <tile />

XmlReader doesn't generate EndElement's for empty elements, so the code was going crazy.

File: TiledLib/Map.cs [hunk 3]
```diff
@@ -38,6 +38,9 @@ public class Map : IXmlSerializable
         [JsonProperty("tileheight")]
         public int CellHeight { get; set; }
 
+        [JsonProperty("hexsidelength")]
+        public int? HexSideLength { get; set; }
```
Comment (BorisTheBrave) [inline TiledLib/Map.cs:42]: I chose to make these new elements nullable as it seems that Tiled doesn't generate them if they are not relevant.

**Comment by Ragath** [comment]
Looks clean and complete, thanks. I'll have to add a map containing some hexagonal layouts to the tests first, but I expect to merge this after that.

**Comment by BorisTheBrave** [comment]
I used `hexagonal-mini.tmx` and `desert.tmx` from the examples that come with Tiled to test these.

**Comment by BorisTheBrave** [comment]
I've added a test for you, along with a few more features necessary to make the test work.

**Comment by Ragath** [comment]
Thanks for the additions.

**Pull request merged**